drivers/led/blink_fallback.c

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2025 BayLibre SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/led.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+
+struct led_blink_fallback_data {
+	const struct device *dev;
+	uint32_t led;
+	struct k_work_delayable work;
+	uint32_t delay_on;
+	uint32_t delay_off;
+};
+
+static void led_blink_fallback_off(struct k_work *work);
+
+static K_MUTEX_DEFINE(led_blink_fallback_mutex);
```

Comment:
use a semaphore, lighter implementation, this does not need the extra stuff that comes with mutextes. https://youtu.be/p8OgqVQxklo?si=z8xJ5mp_Qi86zO9V&t=1305

Reply:
I use static allocation now so this should not be needed anymore.